src/main/java/com/icthh/xm/tmf/ms/communication/messaging/handler/SmppMessagingHandler.java

```diff
@@ -106,13 +160,43 @@ private void sendMessage(MessageResponse messageResponse, String topic) {
         channelResolver.send(topic, messageResponse);
     }
 
+    private Integer getFromCharacteristics(List<CommunicationRequestCharacteristic> characteristic,
+                                           String key, Function<String, Integer> parseFunction) {
+        return Optional.ofNullable(characteristic)
+            .flatMap(c -> findValue(key, c))
+            .map(parseFunction)
+            .orElse(null);
+    }
+
+    private Optional<String> findValue(String key, List<CommunicationRequestCharacteristic> c) {
+        return c.stream()
+            .filter(ch -> key.equals(ch.getName()))
+            .findFirst()
+            .map(CommunicationRequestCharacteristic::getValue);
+    }
+
```

Comment:
@VictorKhoroshko 
the same here, i can propose a way to describe full logic without any additional methods using flatMap
```
    private Integer getFromCharacteristics(List<CommunicationRequestCharacteristic> characteristic,
                                           String key, Function<String, Integer> parseFunction) {
        return Optional.ofNullable(characteristic)
                       .stream()
                       .flatMap(Collection::stream)
                       .filter(ch -> key.equals(ch.getName()))
                       .findFirst()
                       .map(CommunicationRequestCharacteristic::getValue)
                       .map(parseFunction)
                       .orElse(null);
    }
                       ```